### PR TITLE
Stp 1141 date field in edit conditions

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

- Adjust Conditions Date Field to Follow Day/Month/Year format
- Implement `ErrorMessage` component to reduce code repetition and refactor accordingly
- Implement missing inline-conditions name validation

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have manually tested the application to confirm the changes are working as expected. 

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts
- [X] I have checked deployments are working in all environments
- [X] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Review by jenbutongit — APPROVED** [review]
LGTM! maybe raise at a huddle if we should switch over to snapshot testing / snapshot test from now on

File: designer/client/conditions/AbsoluteDateValues.tsx [hunk 1]
```diff
@@ -19,76 +20,125 @@ interface Props {
...
       updateValue({ year: parsedYear, month: parsedMonth, day: parsedDay });
     }
   }, [year, month, day]);
 
-  const yearChanged = (e) => setYear(e.target.value);
-  const monthChanged = (e) => setMonth(e.target.value);
-  const dayChanged = (e) => setDay(e.target.value);
+  const dayChanged = (e: ChangeEvent<HTMLInputElement>) => {
+    const day = e.target.value;
+    if (Number(day) <= 31) {
```
Comment (jonnywyatt) [inline designer/client/conditions/AbsoluteDateValues.tsx:85]: Do you need this, if the number input has a max value of 31?

**Comment by jonnywyatt** [comment]
Snapshot testing is notoriously fragile and includes lots of implementation detail which goes against the point of react testing library and causes lots of false negatives; the volume of snapshots makes them hard to review. Do you see any big advantages in going down the snapshot route?
Some thoughts from [KCD here](https://kentcdodds.com/blog/effective-snapshot-testing)

File: designer/client/conditions/AbsoluteDateValues.tsx (hunk 1 above)
Comment (bertrandmc) [reply designer/client/conditions/AbsoluteDateValues.tsx:85]: QA was able to break it, and manually testing it I am also able to insert any number in days and months using Chrome, max prop is being totally ignored.

**Comment by bertrandmc** [comment]
> Snapshot testing is notoriously fragile and includes lots of implementation detail which goes against the point of react testing library and causes lots of false negatives; the volume of snapshots makes them hard to review. Do you see any big advantages in going down the snapshot route?
> Some thoughts from [KCD here](https://kentcdodds.com/blog/effective-snapshot-testing)

Nope really, I shound't have push it.

**Review by jenbutongit — APPROVED** [review]

File: designer/client/conditions/AbsoluteDateValues.tsx (hunk 1 above)
Comment (jonnywyatt) [reply designer/client/conditions/AbsoluteDateValues.tsx:85]: 👍

**Pull request merged**